server/controllers/finance/reports/monthlyBalance/index.js

```diff
@@ -0,0 +1,176 @@
+
+const q = require('q');
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const util = require('../../../../lib/util');
+const Tree = require('../../../../lib/Tree');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const fiscal = require('../../fiscal');
+
+const TEMPLATE = './server/controllers/finance/reports/monthlyBalance/report.handlebars';
+
+exports.document = document;
+exports.formatData = formatData;
+exports.reporting = reporting;
+
+const DECIMAL_PRECISION = 2; // ex: 12.4567 => 12.46
+
+/**
+ * @description this function helps to get html document of the report in server side
+ * so that we can use it with others modules on the server side
+ * @param {*} options the report options
+ * @param {*} session the session
+ */
+function reporting(opts, session) {
+  const params = opts;
+  params.currency_id = 2;
+  params.allAccount = parseInt(params.allAccount, 10);
+  const accountNumber = params.allAccount ? `` : params.accountNumber;
+  const accountLabel = params.allAccount ? `` : params.accountLabel;
+
+  let docReport;
+  const options = _.extend(opts, {
+    filename : 'FORM.LABELS.MONTHLT_ACCOUNT_ANALYSIS',
+    csvKey : 'rows',
+    user : session.user,
+  });
+
+  try {
+    docReport = new ReportManager(TEMPLATE, session, options);
+  } catch (e) {
+    throw e;
+  }
+
+  let queries;
+  let range;
+
+  const periods = {
+    periodFrom : params.period_id,
+    periodTo : params.period_id,
+  };
+
+  return fiscal.getDateRangeFromPeriods(periods).then(dateRange => {
+    range = dateRange;
+    const sqlParams = [
+      params.fiscal_id,
+      params.period_id,
+    ];
+
+    let filterByAccount;
+    if (accountNumber) {
+      filterByAccount = selectAccountParent(accountNumber);
+    } else {
+      filterByAccount = ``;
```

Comment:
For your information, back quotes ```` are used for multi lines or if we want to insert a variable or an expression : 
```js
`My text and the value of the variable is ${myVariable}` 
```
or in multi-lines : 
```js
`
Line 1
Line 2
etc.
`
```
In this case, you can just write `filterByAccount = ''`;